[GTK] Implement WebAutomationSession::platformGetBase64EncodedPNGData on GTK4 and Skia

TingPing opened the pull request:
#### 6a7c5c176a3a73c1395a92928a20244e68502056
<pre>
[GTK] Implement WebAutomationSession::platformGetBase64EncodedPNGData on GTK4 and Skia
<a href="https://bugs.webkit.org/show_bug.cgi?id=271985">https://bugs.webkit.org/show_bug.cgi?id=271985</a>

Reviewed by Adrian Perez de Castro.

For the variant of platformGetBase64EncodedPNGData that takes a ShareableBitmap
this adds a Skia implementation.

For the variant that takes a ViewSnapshot a GTK4 implementation was added.

* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
* Source/WebKit/UIProcess/Automation/cairo/WebAutomationSessionCairo.cpp:
(WebKit::WebAutomationSession::platformGetBase64EncodedPNGData):
* Source/WebKit/UIProcess/Automation/gtk/WebAutomationSessionGtk.cpp:
(WebKit::base64EncodedPNGData):
(WebKit::WebAutomationSession::platformGetBase64EncodedPNGData):
* Source/WebKit/UIProcess/Automation/skia/WebAutomationSessionSkia.cpp: Copied from Source/WebKit/UIProcess/Automation/cairo/WebAutomationSessionCairo.cpp.
(WebKit::base64EncodedPNGData):
(WebKit::WebAutomationSession::platformGetBase64EncodedPNGData):

Canonical link: <a href="https://commits.webkit.org/276982@main">https://commits.webkit.org/276982@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7f26b07d80bde2bbb25f504a2c42e753c6e9f2ca

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46240 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25372 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48826 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48912 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42281 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29816 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22829 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37776 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46818 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/22518 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39869 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18989 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/19890 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4284 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/42613 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41320 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50733 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/21241 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17736 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/45000 "Found 2 new test failures: imported/w3c/web-platform-tests/trusted-types/Node-multiple-arguments.html, imported/w3c/web-platform-tests/trusted-types/block-Node-multiple-arguments.html (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22536 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/43905 "Passed tests") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/22903 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6466 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22231 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebKit/PlatformWPE.cmake:
```diff
@@ -454,6 +454,12 @@ if (USE_CAIRO)
 
+if (USE_SKIA)
+    list(APPEND WebKit_SOURCES
+        UIProcess/Automation/skia/WebAutomationSessionSkia.cpp
```
Is there some reason to not list this in `SourcesWPE.txt` instead?

TingPing replied:
No reason, sounds good.

carlosgcampos commented on Source/WebKit/UIProcess/Automation/skia/WebAutomationSessionSkia.cpp:
```diff
@@ -0,0 +1,72 @@
+    if (!data)
+        return std::nullopt;
+
+    return base64EncodeToString(std::span<const uint8_t>(data.get()->bytes(), data.get()->size()));
```
I guess you don't need the `.get()` just `data->bytes()` and `data->size()` no?

TingPing committed:
[GTK] Implement WebAutomationSession::platformGetBase64EncodedPNGData on GTK4 and Skia
https://bugs.webkit.org/show_bug.cgi?id=271985

Reviewed by Adrian Perez de Castro.

For the variant of platformGetBase64EncodedPNGData that takes a ShareableBitmap
this adds a Skia implementation.

For the variant that takes a ViewSnapshot a GTK4 implementation was added.

* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
* Source/WebKit/UIProcess/Automation/cairo/WebAutomationSessionCairo.cpp:
(WebKit::WebAutomationSession::platformGetBase64EncodedPNGData):
* Source/WebKit/UIProcess/Automation/gtk/WebAutomationSessionGtk.cpp:
(WebKit::base64EncodedPNGData):
(WebKit::WebAutomationSession::platformGetBase64EncodedPNGData):
* Source/WebKit/UIProcess/Automation/skia/WebAutomationSessionSkia.cpp: Copied from Source/WebKit/UIProcess/Automation/cairo/WebAutomationSessionCairo.cpp.
(WebKit::base64EncodedPNGData):
(WebKit::WebAutomationSession::platformGetBase64EncodedPNGData):

Canonical link: https://commits.webkit.org/276982@main
